cromwell-drs-localizer/src/main/scala/drs/localizer/DrsLocalizerMain.scala

```diff
@@ -132,43 +130,92 @@ object DrsLocalizerMain extends IOApp {
 
 
   private def downloadFileFromGcs(gcsUrl: String,
-                          serviceAccountJsonOption: Option[String],
-                          downloadLoc: String,
-                          requesterPaysProjectIdOption: Option[String]) : IO[Unit] = {
-    logger.info(s"Requester Pays project ID is $requesterPaysProjectIdOption")
-
-    val gcsUrlArray = gcsUrl.replace(GcsScheme, "").split("/", 2)
-    val fileToBeLocalized = gcsUrlArray(1)
-    val gcsBucket = gcsUrlArray(0)
-
-    val unscopedCredentials = serviceAccountJsonOption match {
-      case None => GoogleCredentials.getApplicationDefault()
-      case Some(serviceAccountJson) => GoogleCredentials.fromStream(new ByteArrayInputStream(serviceAccountJson.getBytes()))
+                                  serviceAccountJsonOption: Option[String],
+                                  downloadLoc: String,
+                                  requesterPaysProjectIdOption: Option[String]) : IO[ExitCode] = {
+
+    def setServiceAccount(saJsonPathOption: Option[Path]): String = {
+      saJsonPathOption match {
+        case Some(saJsonPath) =>
+          s"""# Set gsutil to use the service account returned from Martha
+             |gcloud auth activate-service-account --key-file=$saJsonPath > gcloud_output.txt 2>&1
+             |RC_GCLOUD=$$?
+             |if [ "$$RC_GCLOUD" != "0" ]; then
+             |  echo "Failed to activate service account returned from Martha. File won't be downloaded. Error: $$(cat gcloud_output.txt)" >&2
+             |  exit "$$RC_GCLOUD"
+             |else
+             |  echo "Successfully activated service account; Will continue with download. $$(cat gcloud_output.txt)"
+             |fi
+           """.stripMargin
+        case None => ""
+      }
     }
 
-    val credentials = unscopedCredentials.createScoped(List(CloudPlatformAuthScope).asJava)
+    def gcsCopyCommand(flag: String = ""): String = s"gsutil $flag cp $gcsUrl $downloadLoc"
+
+    def recoverWithRequesterPays(): String = {
+      requesterPaysProjectIdOption match {
+        case Some(userProject) =>
+          s"""# Check if error is requester pays. If yes, retry gsutil copy using project flag
+             |if grep -q '$RequesterPaysErrorMsg' gsutil_output.txt; then
+             |  echo "Received 'Bucket is requester pays' error. Attempting again using Requester Pays billing project"
+             |  ${gcsCopyCommand(s"-u $userProject")}
+             |else
+             |  echo "Failed to download the file. Error: $$(cat gsutil_output.txt)" >&2
+             |  exit "$$RC_GSUTIL"
+             |fi""".stripMargin
+        case None =>
+          s"""  echo "Failed to download the file. Error: $$(cat gsutil_output.txt)" >&2
+             |  exit "$$RC_GSUTIL"
+           """.stripMargin
```

Comment:
Replace `""".stripMargin` with `|""".stripMargin`. Otherwise the margin on this line isn't stripped, leaving extra spaces.